dev_guide/app_tutorials/jenkins_pipeline.adoc

```diff
@@ -0,0 +1,185 @@
+// path of the template to use
+def templatePath = 'https://raw.githubusercontent.com/openshift/nodejs-ex/master/openshift/templates/nodejs-mongodb.json'
+// name of the template that will be created
+def templateName = 'nodejs-mongodb-example'
+
+// timeout in 20 minutes if the pipeline hasn't completed
+timeout(time: 20, unit: 'MINUTES') {
+  // use a nodejs jenkins slave
+  node('nodejs') {
+    // select the default cluster
+    openshift.withCluster() {
+      // select the default project
+      openshift.withProject() {
+        echo "Using project: ${openshift.project()}"
+        stage('clean') {
+          // delete everything with this template label
+          openshift.selector("all", [ template : templateName ]).delete()
+          // delete any secrets with this template label
+          if (openshift.selector("secrets", templateName).exists()) {
+            openshift.selector("secrets", templateName).delete()
+          }
+        }
+        stage('create') {
+          // create a new application from the templatePath
+          openshift.newApp(templatePath)
+        }
+        stage('build') {
+          def builds = openshift.selector("bc", templateName).related('builds')
+          // wait up to 5 minutes for the build to complete
+          timeout(5) {
+            builds.untilEach(1) {
+              return (it.object().status.phase == "Complete")
+            }
+          }
+        }
+        stage('deploy') {
+          def rm = openshift.selector("dc", templateName).rollout()
+          // wait up to 5 minutes for the deployment to complete
+          timeout(5) {
+            openshift.selector("dc", templateName).related('pods').untilEach(1) {
+              return (it.object().status.phase == "Running")
+            }
+          }
+        }
+        stage('tag') {
+          // if everything else succeeded, tag the ${templateName}:latest image as ${templateName}:stage
+          // a pipeline build config for the staging environment can watch for the ${templateName}:stage
+          // image to change and then deploy it to the staging environment
+          openshift.tag("${templateName}:latest", "${templateName}:stage")
+        }
+      }
+    }
+  }
+}
+----
+
+[[creating-the-pipeline]]
+
+== Creating the Pipeline
+
+Now we can create the build config in our OpenShift cluster with the following command:
+
+[source]
+----
+$ oc create -f nodejs-sample-pipeline.yaml
+----
+
+If you don't want to create your own file, you can use the sample from the Origin repository with the following command:
+
+[source]
+----
+$ oc create -f https://raw.githubusercontent.com/openshift/origin/master/examples/jenkins/pipeline/nodejs-sample-pipeline.yaml
+----
+
+For more information about the OpenShift DSL syntax used here see https://jenkins.io/doc/pipeline/steps/openshift-pipeline/[OpenShift Pipeline Steps].
+
+[[starting-the-pipeline]]
+
+== Starting the Pipeline
+
+We can now start the pipeline with the following command:
```

Comment:
I'm anticipating the user asking "can I start the pipeline from the Jenkins Console?", which he can of course, and heading off that question with noting that as a footnote type blurb, vs. whether that is a rabbit hole / unnecessary further promotion of Jenkins we want to avoid.

In other words, I'm undecided.  @bparees @coreydaley - what do you guys think?

Reply:
I will update with the three main ways to start it, OpenShift web console, OpenShift CLI, and Jenkins console.

Reply:
updated in next push

Reply:
ok cool